### ISSUE TITLE
TLS 1.3 Compatibility Matrix

### ISSUE THREAD

**Issue by upsampled** [dotnet/docs#35323]
**Issue description**
The documentation for determining if TLS 1.3 should work for a .Net Framework/Core runtime, compile target, and OS combination is dispersed, and incomplete.

Currently most documentation references [this schannel page](https://learn.microsoft.com/en-us/windows/win32/secauthn/protocols-in-tls-ssl--schannel-ssp-#tls-protocol-version-support); however, I seemingly have been able to get dotnet core 6 to use a TLS 1.3 client (the schannel page suggests this is impossible). This might be due to the schannel documentation only applying for 32 bit compile targets (Win32 Apps is a parent page). I am still trying to determine is TLS 1.3 is possible for .NET Framework 4.8

Basically there needs to be something that ties all of this together. I am hoping the document will answer the following:
* What are the dependencies to TLS 1.3 support (dotnet framework/core, OS, compile target)
* Point out the components that determine the dependencies. (IE: schannels are always/sometimes(?) used for the TLS connection, schannels depend on compile target and OS, etc.) 

**Target framework**

- [x] .NET Core
- [x] .NET Framework
- [ ] .NET Standard
